src/main/java/com/conveyal/datatools/manager/models/transform/StringTransformation.java

```diff
@@ -0,0 +1,57 @@
+package com.conveyal.datatools.manager.models.transform;
+
+import com.conveyal.datatools.common.status.MonitorableJob;
+import com.conveyal.datatools.manager.models.TableTransformResult;
+import com.conveyal.datatools.manager.models.TransformType;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+
+public class StringTransformation extends ZipTransformation {
+
+    public static StringTransformation create(String csvData, String table) {
+        StringTransformation transformation = new StringTransformation();
+        transformation.csvData = csvData;
+        transformation.table = table;
+        return transformation;
+    }
+
+    @Override
+    public void validateParameters(MonitorableJob.Status status) {
+        if (csvData == null) {
+            status.fail("CSV data must not be null");
+        }
+    }
+
+    @Override
+    public void transform(FeedTransformZipTarget zipTarget, MonitorableJob.Status status) {
+        String tableName = table + ".txt";
+        Path targetZipPath = Paths.get(zipTarget.gtfsFile.getAbsolutePath());
+        try ( FileSystem targetZipFs = FileSystems.newFileSystem(targetZipPath, (ClassLoader) null) ){
+            // Convert csv data to input stream.
+            InputStream inputStream = new ByteArrayInputStream(csvData.getBytes(StandardCharsets.UTF_8));
+            Path targetTxtFilePath = getTablePathInZip(tableName, targetZipFs);
+            // Set transform type according to whether target file exists.
+            TransformType type = Files.exists(targetTxtFilePath)
+                    ? TransformType.TABLE_REPLACED
+                    : TransformType.TABLE_ADDED;
+            // Copy csv input stream into the zip file, replacing it if it already exists.
+            Files.copy(inputStream, targetTxtFilePath, StandardCopyOption.REPLACE_EXISTING);
+            // Added or updated Count is the number of lines in the CSV.
+            int NEW_LINE_CHARACTER_CODE = 10;
```

Comment:
Done in https://github.com/ibi-group/datatools-server/pull/536/commits/19face8619d29a6c495202c37fa708fe65508ce1

Reply:
@miles-grant-ibigroup this comment here conflicts with your review. Can I let you guys hash it out? To me the all caps constant style makes sense and we use it elsewhere in the code (e.g. http://github.com/ibi-group/datatools-server/blob/42cb12823f379503b060dbadbdbbe49250184454/src/main/java/com/conveyal/datatools/editor/controllers/api/EditorController.java#L61-L61)

Reply:
@br648

Reply:
@philip-cline: You could make the variable final and keep `NEW_LINE_CHARACTER_CODE`. This has been flagged by SonarLint because without the `final` it doesn't conform to a local variable naming standard. The example you have provided is also `final`.

cc: @miles-grant-ibigroup